web-dashboard/src/app/contributors/contributors.component.ts

```diff
@@ -1,16 +1,60 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit, OnDestroy, ElementRef } from '@angular/core';
 import { UserStats } from '../reports/user-stats';
+import { Subscription, Observable } from 'rxjs';
 
 @Component({
   selector: 'app-contributors',
   templateUrl: 'contributors.component.html',
   styleUrls: ['contributors.component.scss']
 })
-export class ContributorsComponent {
+export class ContributorsComponent implements OnInit, OnDestroy {
+
+  static ROTATE_RATE_IN_MILLISECONDS = 3 * 1000;
+
+  private CONTRIBUTOR_POSITION = '--contributor-position';
+  private CONTRIBUTOR_QTY = '--contributor-qty';
+
+  animation: Subscription;
+  position: number;
+  element;
 
   @Input() contributors: Array<UserStats>;
 
-  constructor() {
+  constructor(element: ElementRef) {
+    this.element = element;
+    this.position = 1;
+  }
+
+  ngOnInit(): void {
+    this.animation = Observable
+      .timer(ContributorsComponent.ROTATE_RATE_IN_MILLISECONDS, ContributorsComponent.ROTATE_RATE_IN_MILLISECONDS)
+      .subscribe(() => {
+        this.slideToNextPosition();
+      });
+  }
+
+  private slideToNextPosition() {
+    const visibleContributorQuantity = this.getVisibleContributorQuantity();
+    if (this.contributors.length - this.position >= visibleContributorQuantity) {
+      this.position += 1;
+    } else {
+      this.position = 1;
```

Comment:
No, it has left = contributor-margin, since the second part of the addition is zero-ed by `position - 1`.